src/eval.rs

```diff
@@ -774,6 +774,16 @@ fn reduce_with_witness<F: LurkField>(
                     } else {
                         Control::Return(arg1, env, store.intern_cont_unop(Op1::Char, cont))
                     }
+                } else if head == store.sym("eval") {
+                    let (arg1, end) = match store.car_cdr_mut(&rest) {
+                        Ok((car, cdr)) => (car, cdr),
```

Comment:
I'm curious why you destructure and restructure here instead of 
```rust
Ok(r) => r,
```

Reply:
To the first question, most simply: I just repeated the pattern used everywhere else in this section of the source. As to why it's like that elsewhere, writing it with explicit (named) destructuring makes very clear what is being assigned. A lot of this code is hard to keep track of as a whole, so the more local context made available, the better for the reader (future self or other).

To the second question, note that the `Err(e)` case has an early return from the outer function, so your rewrite is not equivalent.

Reply:
Ok, I am still learning rust so i am trying to get a sense of what is proper.

I definitely am not a fan of asking people to fix in a PR something that was already broke but not related to the PR (which is the case here, even though your code is new, it's part of an existing pattern). So I am not suggesting you need to change anything.

I think maybe you can still use `or_else` with a properly placed `?` but I'll have to try it myself.

Reply:
Yeah, you might be write about the `or_else` if amended. Feel free to make a small PR changing and/or cleaning up the rest of this file's inherited usage if the end result is clearer than what we have now.